st2common/st2common/models/db/execution.py

```diff
@@ -141,11 +152,27 @@ def mask_secrets(self, value):
                         p: "string" for p in liveaction["parameters"]["response"]
                     },
                 )
-
+        
+        output_schema_result = ActionExecutionDB.result.parse_field_value(
+            result["result"]
```

Comment:
The result["result"] here is specific to python action. For shell or workflow action, this is going to return key error. There's an output key defined in the action runner model that says which attribute the action is output to in the result (i.e. result for python, output for workflow, etc.). The output key is defined at https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/runner.py#L70.  You can see how it is used to validate output schema at https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/action_db.py#L247. The mask secret for output will also need to do this. This also means you will need unit tests for result of different action types.